adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -88,79 +88,98 @@ - (void)renderNativeAdForAdConfiguration:
                  strongSelf->_nativeAd = nativeAd;
                  strongSelf->_nativeAd.delegate = strongSelf;
                  strongSelf->_nativeAd.rootViewController = adConfiguration.topViewController;
-
-                 if (strongSelf->_loadCompletionHandler) {
-                   id<GADMediationNativeAdEventDelegate> delegate =
-                       strongSelf->_loadCompletionHandler(strongSelf, nil);
-                   strongSelf->_delegate = delegate;
-                 }
+      
+                 [strongSelf loadRequireData];
                }];
 }
 
-#pragma mark - GADMediationNativeAd
-
-- (GADNativeAdImage *)icon {
-  if (!_icon) {
-    if (_nativeAd.data.icon && _nativeAd.data.icon.imageURL != nil) {
-      _icon = [self imageWithUrlString:_nativeAd.data.icon.imageURL];
+- (void)loadRequireData {
+    NSString *urlString = _nativeAd.data.icon.imageURL;
+    if (!urlString.length) {
+        if (_loadCompletionHandler) _delegate = _loadCompletionHandler(self,nil);
```

Comment:
Rather than needing to wrap this in every location, we can create a block locally at the top like:

```
localBlock = ^{(NSError *error)if (_loadCompletionHandler) {
   _loadCompletionHandler(self,error);
}};
```
then everywhere below can call localBlock without a guard and not risk crashing.

Reply:
Ok~